venia-integration-tests/src/tests/wishList/integrationTests/verifyMultipleWishlistFeatures.spec.js

```diff
@@ -0,0 +1,227 @@
+const { createWishlist } = wishlistPageActions;
+const {
+    assertEmptyWishlistPage,
+    assertEmptyWishlistExists,
+    assertCreateWishlistLink,
+    assertProductInWishlist
+} = wishlistAssertions;
+const { categoryTops, productCarinaCardigan } = categoryPageFixtures;
+const {
+    addProductToWishlistFromCategoryPage,
+    createWishlistViaDialog
+} = categoryPageActions;
+const { assertWishlistSelectedProductOnCategoryPage } = categoryPageAssertions;
+const {
+    productAugustaEarningsUrl,
+    productAugustaEarningsName
+} = productPageFixtures;
+const {
+    addProductToWishlistFromProductPage,
+    addProductToExistingWishlistFromDialog
+} = productPageActions;
+const { assertProductSelectIndicator } = productPageAssertions;
+const {
+    getCustomerWishlistCall,
+    getMultipleWishlistConfigCall,
+    getWishlistDialogDataCall,
+    getWishlistitemsForLocalFieldsCall,
+    hitGraphqlPath,
+    getWishlistConfigForGalleryCall,
+    getWishlistConfigForProductPageCall
+} = graphqlMockedCallsFixtures;
+
+// TODO add tags CE, EE to test to filter and run tests as needed
+describe('verify single wishlist basic features', () => {
+    it('user should be able to add and remove products from wishlist', () => {
+        cy.visitPage(homePage);
+
+        cy.openLoginDialog();
+        cy.createAccount(
+            accountAccessFixtures.firstName,
+            lastName,
+            accountEmail,
+            accountPassword
+        );
+
+        assertCreateAccount(firstName);
+
+        cy.intercept('GET', getCustomerWishlistCall, {
+            fixture: 'wishlist/multipleWishlist/noWishlistPage.json'
+        }).as('getCustomerWishlist');
+        cy.intercept('GET', getMultipleWishlistConfigCall, {
+            fixture: 'wishlist/multipleWishlist/multipleWishlistEnabled.json'
+        }).as('getWishlistConfig');
+        cy.visitPage(wishistRoute);
+        cy.wait(['@getCustomerWishlist']).its('response.body');
+        cy.wait(['@getWishlistConfig']).its('response.body');
+
+        assertEmptyWishlistPage();
+        assertCreateWishlistLink();
+
+        cy.intercept('POST', hitGraphqlPath, req => {
+            if (req.body.operationName.includes('createWishlist')) {
+                req.reply({
+                    fixture:
+                        'wishlist/multipleWishlist/wishlistPageCreateWishlist1.json'
+                });
+            }
+        }).as('createWishlist1');
+        cy.intercept('GET', getCustomerWishlistCall, {
+            fixture: 'wishlist/multipleWishlist/oneWishlistNoProductsPage.json'
+        }).as('getCustomerWishlist1');
+        createWishlist('Test List1');
+        cy.wait(['@getCustomerWishlist1']).its('response.body');
+        cy.wait('@createWishlist1').should(result => {
+            expect(result.request.body.variables).to.eql({
+                input: { name: 'Test List1', visibility: 'PRIVATE' }
+            });
+            expect(result.response.body).to.exist;
+        });
```

Comment:
This test is described as "being able to add and remove products from wishlist", but this block of code creates an empty wishlist and asserts things about it (name, visibility).

It's fine if we have to create a wishlist as part of "being able to add and remove products from wishlist", but I would expect that to be included in something like the `createWishlist` helper function.

If we're asserting that a wishlist is created properly I'd expect that to happen in a separate test case focused specifically on that purpose.

Reply:
I do realize I'm coming from a Unit Testing frame of mind here, so if integration tests are intended to "test everything about wishlists" in one test, we should at least name them as such and add comments for the different blocks / ideas / areas we're testing:

```
it('wishlist page should work', () => {
  // we can create a wishlist
  // ... relevant code and assertions ...
  
  // we can add products to a wishlist
  // ... relevant code and assertions ...

   // we can remove products to a wishlist
  // ... relevant code and assertions ...
  
  // etc.
});
```

Reply:
Yes, for sure I can add comments to clarify.   It's always good to keep test simple and short when it comes to mocked data tests. But in this case, I was unable to mock account creation and to test wishlist we have to be within active session. So, to avoid account creation for every small of piece of functionality I have combined all in one.